dash/_callback.py

```diff
@@ -462,10 +489,19 @@ def add_context(*args, **kwargs):
                 if output_value is callback_manager.UNDEFINED:
                     return to_json(response)
             else:
-                output_value = _invoke_callback(func, *func_args, **func_kwargs)
-
-            if NoUpdate.is_no_update(output_value):
-                raise PreventUpdate
+                try:
+                    output_value = _invoke_callback(func, *func_args, **func_kwargs)
+                except PreventUpdate as err:
+                    raise err
+                except Exception as err:  # pylint: disable=broad-exception-caught
+                    if error_handler:
+                        error_handler(err)
+                        if not multi:
+                            output_value = NoUpdate()
+                        else:
+                            output_value = [NoUpdate for _ in output_spec]
```

Comment:
```suggestion
                    if error_handler:
                        output_value = error_handler(err)
                        if not multi:
                            output_value = output_value or NoUpdate()
                        else:
                            output_value = output_value or [NoUpdate() for _ in output_spec]
```

@T4rk1n I feel that we should still allow the dev to return outputs from the error handler -- we should supply the `NoUpdate()` outputs _only_ if the error handler doesn't return anything. (I realize the dev could still update the outputs via `set_props()`, but I don't see any downside to also allowing them to output the "normal" way.) Happy to discuss further if you disagree.

Reply:
For this wouldn't the dev just handle the error inside of the callback itself, as no two outputs are going to be the same?

Also, I thought that this would return a regular output? The no_update seems to be only supplied if there was no returned value?

Reply:
Doing it like this could lead to situation where you wanted to output something that evaluate to false but it would get a `no_update` instead. see https://github.com/plotly/dash/issues/2906#issuecomment-2209086283, a components which has no children props will always be false.

Best we could do is `if output_value is not None`, but that brings back the age old discussion of undefined vs none, which python lacks an undefined type and None could be a valid prop.

Reply:
What about this:
```
                    if error_handler:
                        output_value = error_handler(err)
                        if output_value is not None or output_value is False:
                            # Code to execute if a value is returned
                            '''output no change'''
                        else:
                            # Code to execute if no value is returned
                            output_value = [NoUpdate() for _ in output_spec] if multi else NoUpdate()
                            ```

Then have it documented that this will pass `NoUpdate` in error if you are really trying to pass a `None` as a single output. If this is the case, then the dev should instead place: `[None]`

Reply:
> For this wouldn't the dev just handle the error inside of the callback itself, as no two outputs are going to be the same?

For the global error handler, yes, this is true -- I can't imagine a situation where you'd want to return specific outputs from the global error handler. But for a callback-specific error handler the dev might want to return specific outputs.

> python lacks an undefined type and `None` could be a valid prop.

Ugh, this is a good point. Thanks Python. I think I would still be OK with the solution of checking `if output_value is not None`, and then documenting that in the specific edge case where you do actually want to return `None`, you have to use `set_props()`.

Reply:
> callback-specific error handler

unless they are reusing, they could still handle inside of the callback, but our syntax is still easier I guess, haha. Rather than them having to wrap the whole function in a try except clause.

Reply:
Changed to NoOutput only if `output_value` is None, the output could be set to `[None]` if the dev want to set the output prop to null. cc @LiamConnors

Reply:
> the output could be set to [None] if the dev want to set the output prop to null.

FWIW, does this actually work? Typically I have gotten errors when wrapping a single output in a list.

Reply:
Yes, it works, but only would work if the output definition in the callback is `[Output(id, prop)]`